[BUG] map_test failures on Spark 3.3.0

Two tests in map_test failed in last night's run against Spark 3.3.0:
```
09:13:42  FAILED ../../src/main/python/map_test.py::test_simple_get_map_value_with_strict_index[Map(String(not_null),String)-true]
09:13:42  FAILED ../../src/main/python/map_test.py::test_map_element_at_ansi_fail[Map(String(not_null),String)]
```

Details:
```
09:13:42  =================================== FAILURES ===================================
09:13:42  _ test_simple_get_map_value_with_strict_index[Map(String(not_null),String)-true] _
09:13:42  
09:13:42  strict_index = 'true', data_gen = Map(String(not_null),String)
09:13:42  
09:13:42      @pytest.mark.skipif(is_before_spark_330(),
09:13:42                          reason="Only in Spark 3.3.0 + ANSI mode + Strict Index, map key throws on no such element")
09:13:42      @pytest.mark.parametrize('strict_index', ['true', 'false'])
09:13:42      @pytest.mark.parametrize('data_gen', [simple_string_to_string_map_gen], ids=idfn)
09:13:42      def test_simple_get_map_value_with_strict_index(strict_index, data_gen):
09:13:42          message = "org.apache.spark.SparkNoSuchElementException"
09:13:42          test_conf=copy_and_update(ansi_enabled_conf, {'spark.sql.ansi.strictIndexOperator': strict_index})
09:13:42          if strict_index == 'true':
09:13:42  >           assert_gpu_and_cpu_error(
09:13:42                      lambda spark: unary_op_df(spark, data_gen).selectExpr(
09:13:42                              'a["NOT_FOUND"]').collect(),
09:13:42                      conf=test_conf,
09:13:42                      error_message=message)
09:13:42  
09:13:42  ../../src/main/python/map_test.py:297: 
09:13:42  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
09:13:42  ../../src/main/python/asserts.py:572: in assert_gpu_and_cpu_error
09:13:42      assert_py4j_exception(lambda: with_cpu_session(df_fun, conf), error_message)
09:13:42  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
09:13:42  
09:13:42  func = <function assert_gpu_and_cpu_error.<locals>.<lambda> at 0x7fa2a5272040>
09:13:42  error_message = 'org.apache.spark.SparkNoSuchElementException'
09:13:42  
09:13:42      def assert_py4j_exception(func, error_message):
09:13:42          """
09:13:42          Assert that a specific Java exception is thrown
09:13:42          :param func: a function to be verified
09:13:42          :param error_message: a string such as the one produce by java.lang.Exception.toString
09:13:42          :return: Assertion failure if no exception matching error_message has occurred.
09:13:42          """
09:13:42          with pytest.raises(Py4JJavaError) as py4jError:
09:13:42              func()
09:13:42  >       assert error_message in str(py4jError.value.java_exception)
09:13:42  E       AssertionError
09:13:42  
09:13:42  ../../src/main/python/asserts.py:561: AssertionError
09:13:42  _________ test_map_element_at_ansi_fail[Map(String(not_null),String)] __________
09:13:42  
09:13:42  data_gen = Map(String(not_null),String)
09:13:42  
09:13:42      @pytest.mark.parametrize('data_gen', [simple_string_to_string_map_gen], ids=idfn)
09:13:42      def test_map_element_at_ansi_fail(data_gen):
09:13:42          message = "org.apache.spark.SparkNoSuchElementException" if (not is_before_spark_330() or is_databricks104_or_later()) else "java.util.NoSuchElementException"
09:13:42          # For 3.3.0+ strictIndexOperator should not affect element_at
09:13:42          test_conf=copy_and_update(ansi_enabled_conf, {'spark.sql.ansi.strictIndexOperator': 'false'})
09:13:42  >       assert_gpu_and_cpu_error(
09:13:42                  lambda spark: unary_op_df(spark, data_gen).selectExpr(
09:13:42                      'element_at(a, "NOT_FOUND")').collect(),
09:13:42                      conf=test_conf,
09:13:42                      error_message=message)
09:13:42  
09:13:42  ../../src/main/python/map_test.py:366: 
09:13:42  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
09:13:42  ../../src/main/python/asserts.py:572: in assert_gpu_and_cpu_error
09:13:42      assert_py4j_exception(lambda: with_cpu_session(df_fun, conf), error_message)
09:13:42  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
09:13:42  
09:13:42  func = <function assert_gpu_and_cpu_error.<locals>.<lambda> at 0x7fa2a52c5160>
09:13:42  error_message = 'org.apache.spark.SparkNoSuchElementException'
09:13:42  
09:13:42      def assert_py4j_exception(func, error_message):
09:13:42          """
09:13:42          Assert that a specific Java exception is thrown
09:13:42          :param func: a function to be verified
09:13:42          :param error_message: a string such as the one produce by java.lang.Exception.toString
09:13:42          :return: Assertion failure if no exception matching error_message has occurred.
09:13:42          """
09:13:42          with pytest.raises(Py4JJavaError) as py4jError:
09:13:42              func()
09:13:42  >       assert error_message in str(py4jError.value.java_exception)
09:13:42  E       AssertionError
09:13:42  
09:13:42  ../../src/main/python/asserts.py:561: AssertionError
```

--- a/integration_tests/src/main/python/map_test.py
+++ b/integration_tests/src/main/python/map_test.py
@@ -288,7 +288,7 @@ def test_simple_get_map_value_ansi_fail(data_gen):
 
 @pytest.mark.skipif(is_before_spark_330(),
                     reason="Only in Spark 3.3.0 + ANSI mode + Strict Index, map key throws on no such element")
-@pytest.mark.skipif(not is_before_spark_330(),
+@pytest.mark.xfail(not is_before_spark_330(),
                     reason="There was a bug introduced in Spark 3.3.0 which is being tracked by " \
                            "https://issues.apache.org/jira/browse/SPARK-39015")
 @pytest.mark.parametrize('strict_index', ['true', 'false'])
@@ -362,7 +362,7 @@ def test_element_at_map_timestamp_keys(data_gen):
 
 
 @pytest.mark.parametrize('data_gen', [simple_string_to_string_map_gen], ids=idfn)
-@pytest.mark.skipif(not is_before_spark_330(),
+@pytest.mark.xfail(not is_before_spark_330(),
                     reason="There was a bug introduced in Spark 3.3.0 which is being tracked by " \
                            "https://issues.apache.org/jira/browse/SPARK-39015")
 def test_map_element_at_ansi_fail(data_gen):

--- a/integration_tests/src/main/python/map_test.py
+++ b/integration_tests/src/main/python/map_test.py
@@ -288,6 +288,9 @@ def test_simple_get_map_value_ansi_fail(data_gen):
 
 @pytest.mark.skipif(is_before_spark_330(),
                     reason="Only in Spark 3.3.0 + ANSI mode + Strict Index, map key throws on no such element")
+@pytest.mark.skipif(not is_before_spark_330(),
+                    reason="There was a bug introduced in Spark 3.3.0 which is being tracked by " \
+                           "https://issues.apache.org/jira/browse/SPARK-39015")
 @pytest.mark.parametrize('strict_index', ['true', 'false'])
 @pytest.mark.parametrize('data_gen', [simple_string_to_string_map_gen], ids=idfn)
 def test_simple_get_map_value_with_strict_index(strict_index, data_gen):
@@ -359,6 +362,9 @@ def test_element_at_map_timestamp_keys(data_gen):
 
 
 @pytest.mark.parametrize('data_gen', [simple_string_to_string_map_gen], ids=idfn)
+@pytest.mark.skipif(not is_before_spark_330(),
+                    reason="There was a bug introduced in Spark 3.3.0 which is being tracked by " \
+                           "https://issues.apache.org/jira/browse/SPARK-39015")
 def test_map_element_at_ansi_fail(data_gen):
     message = "org.apache.spark.SparkNoSuchElementException" if (not is_before_spark_330() or is_databricks104_or_later()) else "java.util.NoSuchElementException"
     # For 3.3.0+ strictIndexOperator should not affect element_at

